Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Net.Mail;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Creates new <see cref="Uri"/> instances.
+    /// </summary>
+    public class MailAddressGenerator : ISpecimenBuilder
+    {
+        private readonly string[] fictitiousDomains = 

```

Comment:
While these predefined domains correctly follow [this](https://github.com/AutoFixture/AutoFixture/issues/339#issuecomment-70984249) comment by @ploeh, wouldn't it be more flexible if we could also allow users to supply their own domains if they need to?

A similar example is the `UriGenerator` class, which [requests a `UriScheme`] (https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/UriGenerator.cs#L31) from the context.